pkg/ticket/filter.go

```diff
@@ -412,74 +429,100 @@ func GetFilter(filterName string, debugFlag bool) (Filter, error) {
 	if err != nil {
 		return Filter{}, err
 	}
+	// Does filters have key named filterName?
+	if _, ok := filters.Filters[filterName]; !ok {
+		return Filter{}, fmt.Errorf("Filter not found: " + filterName)
+	}
 	return filters.Filters[filterName], nil
 }
 
 // FilterTickets takes a list of tickets, a filter name, and a debug flag. It
 // returns a list of tickets that match the filter. Returns an error if there
 // is one.
 func FilterTickets(tickets []Ticket, filterName string, debugFlag bool) (*[]Ticket, error) {
+	debug.DebugMessage(debugFlag, "Filtering tickets with filter: "+filterName+" on "+strconv.Itoa(len(tickets))+" tickets")
+	var listOfTicketsAsAny []any
+
 	// Get the filter
 	filter, err := GetFilter(filterName, debugFlag)
 	if err != nil {
+		debug.DebugMessage(debugFlag, "Error in FilterTickets while getting filter: "+err.Error())
 		return nil, err
 	}
 
+	// If there are no tickets, return
+	if len(tickets) == 0 {
+		return &[]Ticket{}, nil
+	}
+
 	// Parse the filter
 	queryObj, err := gojq.Parse(filter.Filter)
 	if err != nil {
+		debug.DebugMessage(debugFlag, "Error in FilterTickets while parsing filter: "+err.Error())
 		return nil, fmt.Errorf("Error parsing filter: " + err.Error())
 	}
 
-	// Convert []Ticket into []map[string]interface{} for gojq
-	var listOfTickets []map[string]interface{}
+	// Convert []Ticket for gojq
 	ticketsJSON, err := json.Marshal(tickets)
 	if err != nil {
+		debug.DebugMessage(debugFlag, "Error in FilterTickets while marshalling tickets: "+err.Error())
 		return nil, err
 	}
-	fmt.Println("The list of tickets as JSON: " + string(ticketsJSON))
-	err = json.Unmarshal(ticketsJSON, &listOfTickets)
+	debug.DebugMessage(debugFlag, "ticketsJSON: "+string(ticketsJSON))
+	err = json.Unmarshal(ticketsJSON, &listOfTicketsAsAny)
 	if err != nil {
```

Comment:
I feel there's way too much error handling. It interrupts the flow when reading the code. Is this a mandatory golang thing? In other languages, you just let exceptions bubble up to an exception handler.